### PR TITLE
Add group concat function

### PR DESCRIPTION
**Pull request by JuanMGN**
# Description

## Type of change

Make sure you add the correct label for the PR:

- `enhacement` if this PR adds a new feature
- `bug` if it fixes a bug
- `documentation` if it adds docs
- `breaking-change` if this PR introduces a breaking change
- `dependency-updates` if it updates dependencies

### CODE REVIEW

File: modules/engine/src/main/scala/com/gsk/kg/engine/functions/FuncAgg.scala [hunk 1]
```diff
@@ -106,7 +106,6 @@ object FuncAgg {
     )
   }
 
-  // TODO: Implement group-concat https://github.com/gsk-aiops/bellman/issues/202
   def groupConcat(col: Column, separator: String): Column =
-    ???
+    array_join(collect_list(col), separator)
```
Comment (pepegar) [inline modules/engine/src/main/scala/com/gsk/kg/engine/functions/FuncAgg.scala:110]: Lovely @JuanMGN !  I thought we would need to use an UDF at this point!

**Review by pepegar — APPROVED** [review]
GJ!

File: modules/parser/src/main/scala/com/gsk/kg/sparqlparser/AggregateParser.scala [hunk 2]
```diff
@@ -27,14 +27,13 @@ object AggregateParser {
...
+          _ != '''
+        ).! ~ "'" ~ ")" ~
         StringValParser.variable ~
-        "\"" ~
-        CharsWhile(_ != '\"').! ~
-        "\"" ~
         ")"
     )
-      .map(p => Aggregate.GROUP_CONCAT(p._1, p._2))
+      .map(p => Aggregate.GROUP_CONCAT(p._2, p._1))
```
Comment (TomyMeren) [inline modules/parser/src/main/scala/com/gsk/kg/sparqlparser/AggregateParser.scala:36]: why this change?

**Review by TomyMeren — APPROVED** [review]
LGTM!

File: modules/parser/src/main/scala/com/gsk/kg/sparqlparser/AggregateParser.scala (hunk 2 above)
Comment (patofp) [reply modules/parser/src/main/scala/com/gsk/kg/sparqlparser/AggregateParser.scala:36]: I don't know exactly because this was part of the code that @JuanMGN did, but for what I can see the Sparql Algebra switch places between the variable and the separator respect to our case class.

E.g:
```
(group_concat (separator '|') ?v)
```

**Review by patofp — APPROVED** [review]
